src/js/users/components/CreateUserForm.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { Controller, useForm } from "react-hook-form";
+import {
+    Checkbox,
+    InputError,
+    InputGroup,
+    InputLabel,
+    InputSimple,
+    ModalBody,
+    ModalFooter,
+    SaveButton,
+} from "../../base";
+
+interface CreateUserFormProps {
+    /** The user's handle or username */
+    handle?: string;
+    /** The user's password */
+    password?: string;
+    /** Error message to be displayed */
+    error: string;
+    /** A callback function to be called when the form is submitted */
+    onSubmit: (data: { handle: string; password: string; forceReset: boolean }) => void;
+}
+
+/**
+ * A form component for creating a new user
+ */
+export function CreateUserForm({ handle = "", password = "", error, onSubmit }: CreateUserFormProps) {
+    const {
+        formState: { errors },
+        register,
+        handleSubmit,
+        control,
+    } = useForm({ defaultValues: { handle, password, forceReset: false } });
+
+    return (
+        <form onSubmit={handleSubmit(values => onSubmit({ ...values }))}>
+            <ModalBody>
+                <InputGroup>
+                    <InputLabel htmlFor="handle">Username</InputLabel>
+                    <InputSimple id="handle" {...register("handle", { required: "Please specify a username" })} />
+                    <InputError>{errors.handle?.message}</InputError>
+                </InputGroup>
+                <InputGroup>
+                    <InputLabel htmlFor="password">Password</InputLabel>
+                    <InputSimple
+                        id="password"
+                        type="password"
+                        {...register("password", {
+                            required: "Password does not meet minimum length requirement (8)",
+                            minLength: {
+                                value: 8,
+                                message: "Password does not meet minimum length requirement (8)",
+                            },
```

Comment:
Does min-length imply required?, might be able to cut `required`

Reply:
I believe `required` is needed since `minLength` doesn't give a message if the input is empty. Its a bit redundant but we can replace `minLength` with the mutation error instead if that looks cleaner.